### ISSUE TITLE
[rule] inclusive-language

### ISSUE THREAD

**Issue by gabrielcsapo** [eslint/eslint#13417]
**Please describe what the rule should do:**

The rule should identify a list of terms that are not inclusive and offer suggestions for terms that are more inclusive and have similar connotation. 

**What new ECMAScript feature does this rule relate to?**

None.

**What category of rule is this? (place an "X" next to just one item)**

[ ] Warns about a potential error (problem)
[x] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
const blacklist = []; // blacklist is not an inclusive term, the following terms offer similar connotation with a more inclusive inferred meaning. (blocklist, excludelist)

if(badActor) {
 blacklist.push(badActor);
}
```

**Why should this rule be included in ESLint (instead of a plugin)?**

As a main source of linting javascript in the community, having this rule be central to how our community operates and is inclusive would set a precedent for others moving away from terms that are not inclusive and don't offer safe spaces for people to work in.

**Are you willing to submit a pull request to implement this rule?**

Yes.

**Comment by ljharb** [comment]
Not only does this sound like something best not included in core, but you may find https://twitter.com/tesseralis/status/1260421410192650241 to be helpful reading.

**Comment by gabrielcsapo** [comment]
@ljharb certain terms used out of context would still apply to this rule. alexjs tries to blanket terms as "not good". This is not what I am proposing in this rule.

**Comment by ljharb** [comment]
The danger exists with any term; either way, I doubt this would meet eslint's new rules criteria (ie, no new stylistic rules).

**Comment by wooorm** [comment]
Yes, there’s danger, and language *is* super complex, and maybe it doesn’t meet ESLint’s standard (although *stylistic* doesn’t do the problem justice). Still, whether something is complex doesn’t mean one shouldn’t try. I think ESLint could reach a broad audience and it would be super cool if ESLint took a stand on `blacklist`!

**Comment by ljharb** [comment]
To be clear; I think folks should use a meaningful name, and "whitelist" and "blacklist" isn't actually describing what you're using the list for. I just don't think an eslint rule is the best way to make that change happen.
